### ISSUE TITLE
localstorage does not get updated when embedded in an iframe

### ISSUE THREAD

**Issue by Loki-Afro** [RocketChat/Rocket.Chat#23937]
### Description:

Our setup is quite advanced i'd say:
We create create rocketchat-groups by creating a "group" in our software with a super user.
When a "group" is created also the creator of that "group" is the moderator of that newly created rocketchat-group.
So initially every room starts with 2 users the super admin and an moderator - this is all done by api requests.

We embed rocketchat via iframe into our application, it is common to have serveral rocketchat iframes open at the same time in different tabs to chat with the users within that group.

I wanted to update rocketchat from 3.16.3 to 4.2.0 and this issue occurred: "User sees join button but is part of that group already"
It appears that this only happens for the second group a user creates, so i dig deeper into this and found out:
**When using rocketchat within an iframe localstorage rooms/subscriptions might not get updated and therefore user sees join button.**

Update: Also 3.16.3 is affected.

### Steps to reproduce:

1. embed rocketchat as an iframe and happily use it 
2. embed a new room an an iframe - can't use it

it is really hard to reproduce if one has not the same setup to be honest.

### Expected behavior:

**localstorage and all its cached content gets updated when opening a new iframe**

### Actual behavior:

localstorage does not get updated when opening rocketchat in an new iframe

### Server Setup Information:

- Version of Rocket.Chat Server: 4.2.0
- Operating System: docker
- Deployment Method: kubernetes/docker
- Number of Running Instances: 1
- DB Replicaset Oplog: enabled
- NodeJS Version: see in docker
- MongoDB Version: 4.2.9

### Client Setup Information

- Desktop App or Browser Version:
- Operating System: linux / mac - happens in Firefox, could be more

### Additional context
This join button is not wanted at all since that path is blocked on our side, we want as little as possible of user interaction directly with rocketchat.

![image](https://user-images.githubusercontent.com/4738445/145867585-ef8f3dbf-f1bc-4712-8f66-92067f6c0fa7.png)

I think i pinned it down to this line https://github.com/RocketChat/Rocket.Chat/blob/e1c527a594a33e23e4fc33bff3c8525ca37a4ed3/app/ui-message/client/messageBox/messageBox.html#L36
and subsequently the caching layer fails.

Additionally i can add that when using rocketchat in a browser - like it is supposed to be used - with all its functionality this issue is not happening - this seems to be iframe related :/

### Relevant logs:

as i said a quite restricted environment and offsetBla is not white listed but didn't cause any known issues yet
![image](https://user-images.githubusercontent.com/4738445/145868369-3ff64cd6-1b92-4dc6-9323-f5c11d29d62c.png)
```
LocalStore: store created at 
LocalStore: store created at 
LocalStore: store created at 
{"level":51,"time":"2021-12-13T17:27:06.737Z","pid":1,"hostname":"rocketchat-deployment-5548fdc496-4x6kj","name":"Migrations","msg":"Not migrating, already at version 249"}
ufs: temp directory created at "/tmp/ufs"
Loaded the Apps Framework and loaded a total of 0 Apps!
Users with admin role already exist; Ignoring environment variables ADMIN_PASS
+--------------------------------------------------------------------------+
|                              SERVER RUNNING                              |
+--------------------------------------------------------------------------+
|                                                                          |
|  Rocket.Chat Version: 4.2.0                                              |
|       NodeJS Version: 12.22.1 - x64                                      |
|      MongoDB Version: 4.2.9                                              |
|       MongoDB Engine: wiredTiger                                         |
|             Platform: linux                                              |
|         Process Port: 5000                                               |
|             Site URL: xxxx  |
|     ReplicaSet OpLog: Enabled                                            |
|          Commit Hash: a4ad199f13                                         |
|        Commit Branch: HEAD                                               |
|                                                                          |
+--------------------------------------------------------------------------+
```

**Comment by Loki-Afro** [comment]
i can also provide a system where the issue occurs, just pm me for that

**Comment by guijun13** [comment]
It's working properly on the latest version, but if it is still happening to you, please open a new issue with more details. Thank you very much!
